relay-server/src/utils/managed_envelope.rs

```diff
@@ -312,7 +312,11 @@ impl ManagedEnvelope {
         if self.context.summary.profile_quantity > 0 {
             self.track_outcome(
                 outcome,
-                DataCategory::Profile,
+                if self.context.summary.event_metrics_extracted {
+                    DataCategory::ProfileIndexed
+                } else {
+                    DataCategory::Profile
+                },
```

Comment:
I did not add a separate integration test for this change, because https://github.com/getsentry/relay/blob/51888358224631ac085b71bb5dd49e87fab0fb9d/tests/integration/test_outcome.py#L981 already covers it. That test failed after adding the change in this file, and succeeded again after also making the change in `relay-server/src/actors/outcome.rs`.